Build failure: meilisearch

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *X*

### Build log

```
error: builder for '/nix/store/q6wg8k0ig499mmqhy452gs8c96nn48zc-meilisearch-1.9.0.drv' failed with exit code 101;
       last 25 log lines:
       >    Compiling actix-web-codegen v4.2.0
       >    Compiling tempfile v3.10.1
       >    Compiling bincode v1.3.3
       >    Compiling time v0.3.34
       >    Compiling gemm-common v0.17.1
       >    Compiling kstring v2.0.0
       >    Compiling irg-kvariants v0.1.0
       >    Compiling itertools v0.11.0
       > error[E0282]: type annotations needed for `Box<_>`
       >   --> /build/cargo-vendor-dir/time-0.3.34/src/format_description/parse/mod.rs:83:9
       >    |
       > 83 |     let items = format_items
       >    |         ^^^^^
       > ...
       > 86 |     Ok(items.into())
       >    |              ---- type must be known at this point
       >    |
       > help: consider giving `items` an explicit type, where the placeholders `_` are specified
       >    |
       > 83 |     let items: Box<_> = format_items
       >    |              ++++++++
       >
       > For more information about this error, try `rustc --explain E0282`.
       > error: could not compile `time` (lib) due to 1 previous error
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/q6wg8k0ig499mmqhy452gs8c96nn48zc-meilisearch-1.9.0.drv'.
error: 1 dependencies of derivation '/nix/store/w77dplh87rhy52jcf3kchcr19gc1qhvn-unit-meilisearch.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gzijk1ffvyhygwnzk412k797fl91pfiz-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/g15dhd643rza23n12damd3h5dgal3l7i-etc-json.drv' failed to build
error: 1 dependencies of derivation '/nix/store/inkjzhw01phpzzc0fcmgbmawszjkkimr-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/619dssd8sb0my13zasql9yjnbdwsggyz-nixos-system-hastur-24.11.20240824.9edd5dd.drv' failed to build
```

### Additional context

Add any other context about the problem here.

### Notify maintainers
@happysalada
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
nixos-unstable-small
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/servers/search/meilisearch/default.nix
+++ b/pkgs/servers/search/meilisearch/default.nix
@@ -23,6 +23,11 @@ rustPlatform.buildRustPackage {
     hash = "sha256-fPXhayS8OKiiiDvVvBry3njZ74/W6oVL0p85Z5qf3KA==";
   };
 
+  cargoPatches = [
+    # fix build with Rust 1.80
+    ./time-crate.patch
+  ];
+
   cargoBuildFlags = [ "--package=meilisearch" ];
 
   cargoLock = {

--- a/pkgs/servers/search/meilisearch/time-crate.patch
+++ b/pkgs/servers/search/meilisearch/time-crate.patch
@@ -1,0 +1,28 @@
+diff --git a/Cargo.lock b/Cargo.lock
+index 3c728f348..51df0ea7b 100644
+--- a/Cargo.lock
++++ b/Cargo.lock
+@@ -5098,9 +5098,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "time"
+-version = "0.3.34"
++version = "0.3.36"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "c8248b6521bb14bc45b4067159b9b6ad792e2d6d754d6c41fb50e29fefe38749"
++checksum = "5dfd88e563464686c916c7e46e623e520ddc6d79fa6641390f2e3fa86e83e885"
+ dependencies = [
+  "deranged",
+  "itoa",
+@@ -5121,9 +5121,9 @@ checksum = "ef927ca75afb808a4d64dd374f00a2adf8d0fcff8e7b184af886c3c87ec4a3f3"
+ 
+ [[package]]
+ name = "time-macros"
+-version = "0.2.17"
++version = "0.2.18"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "7ba3a3ef41e6672a2f0f001392bb5dcd3ff0a9992d618ca761a11c3121547774"
++checksum = "3f252a68540fde3a3877aeea552b832b40ab9a69e318efd078774a01ddee1ccf"
+ dependencies = [
+  "num-conv",
+  "time-core",

